src/ClientImpl.cpp

```diff
@@ -1255,6 +1282,21 @@ void ClientImpl::run() throw (Exception, NoConnectionsException, LibEventExcepti
     m_loopBreakRequested = false;
 }
 
+static void interrupt_callback(evutil_socket_t fd, short events, void *clientData);
+
+void ClientImpl::runForMaxTime(uint64_t uSec) throw (Exception, NoConnectionsException, LibEventException){
+    struct timeval timeOut;
+    logMessage(ClientLogger::DEBUG, "ClientImpl::runForMaxTime");
+
+    timeOut.tv_sec = uSec / 1000000;
+    timeOut.tv_usec = uSec % 1000000;
+
+    event_base_once(m_base, -1, EV_TIMEOUT, interrupt_callback, this, &timeOut);
```

Comment:
I guess you are now the libevent expert.  

I think what this is saying is: "add an event to m_base that will invoke the interrupt_callback (which I don't know where it's actually defined) when the timeout is triggered."  Because you are creating the event with "event_base_once" the event goes away after it's fired.